osc-server/src/test/java/org/osc/core/broker/service/tasks/conformance/openstack/securitygroup/CreatePortGroupTaskTest.java

```diff
@@ -0,0 +1,302 @@
+
+		this.exception.expect(Exception.class);
+		this.exception.expectMessage(String.format("RegisterNetworkElement failed to return PortGroup"));
+
+		CreatePortGroupTask task = new CreatePortGroupTask(sg);
+
+		// Act.
+		task.execute();
+
+		// Assert.
+		verify(this.em, Mockito.never()).merge(any());
+	}
+
+	private SecurityGroupMember newSGMWithPort(Class<? extends OsProtectionEntity> entityType, Long sgmId) {
+		VMPort port = null;
+		OsProtectionEntity protectionEntity;
+
+		if (entityType == VM.class) {
+			protectionEntity = new VM("region", UUID.randomUUID().toString(), "name");
+			port = newVMPort((VM) protectionEntity);
+		} else if (entityType == Network.class) {
+			protectionEntity = new Network("region", UUID.randomUUID().toString(), "name");
+			port = new VMPort((Network) protectionEntity, "mac-address", UUID.randomUUID().toString(),
+					UUID.randomUUID().toString(), null);
+		} else {
+			protectionEntity = new Subnet("network", UUID.randomUUID().toString(), "name", "region", false);
+			port = new VMPort((Subnet) protectionEntity, "mac-address", UUID.randomUUID().toString(),
+					UUID.randomUUID().toString(), null);
+		}
+
+		this.vmProtectedPorts.add(port);
+		return newSGM(protectionEntity, sgmId);
+	}
+
+	private SecurityGroupMember newSGM(OsProtectionEntity protectionEntity, Long sgmId) {
+		// TODO emanoel: Remove this mock once the SGM is no longer kept in a
+		// TreeSet in the SGM.
+		SecurityGroupMember sgm = Mockito.spy(new SecurityGroupMember(protectionEntity));
+		Mockito.doReturn(-1).when(sgm).compareTo(Mockito.any());
+		sgm.setId(sgmId);
+
+		return sgm;
+	}
+
+	private VMPort newVMPort(VM vm) {
+		return new VMPort(vm, "mac-address" + UUID.randomUUID().toString(), UUID.randomUUID().toString(),
+				UUID.randomUUID().toString(), null);
+	}
+
+	private SecurityGroup createSecurityGroup(Long vcId, String tenantId, String tenantName, Long sgId, String sgName) {
+		VirtualizationConnector vc = new VirtualizationConnector();
+		vc.setId(1L);
+		SecurityGroup sg = new SecurityGroup(vc, tenantId, tenantName);
+		sg.setId(sgId);
+		sg.setName(sgName);
+
+		when(this.em.find(SecurityGroup.class, sg.getId())).thenReturn(sg);
+		return sg;
+	}
+
+	private List<NetworkElement> ostRegisterPorts(SecurityGroup sg) throws Exception {
+		List<NetworkElement> neList = this.vmProtectedPorts.stream().map(NetworkElementImpl::new)
+				.collect(Collectors.toList());
+		PowerMockito.doReturn(neList).when(OpenstackUtil.class, "getPorts",
+				sg.getSecurityGroupMembers().iterator().next());
+		return neList;
+	}
+
+	private void registerNetworkRedirectionApi(SdnRedirectionApi redirectionApi, VirtualizationConnector vc)
+			throws Exception {
+		PowerMockito.spy(SdnControllerApiFactory.class);
+		PowerMockito.doReturn(redirectionApi).when(SdnControllerApiFactory.class, "createNetworkRedirectionApi", vc);
+	}
+
+	private SdnRedirectionApi mockRegisterNetworkElement(NetworkElement ne) throws Exception {
```

Comment:
registerNetworkElement for consistency, the other methods dont have the prefix 'mock'